flutter/flutter
# [Feature] Add support for Bundler to manage Gem dependencies

What's this feature about?
-
Added `--[no]-use-bundler` flag so that you can manage Cocoapods gem using `Bundler` and execute `bundle exec pod install` rather than `pod install` if `Gemfile` exists.

Closes these issues: 
-
1. [[DISCUSSION] Use bundler to configure dependencies for iOS project](https://github.com/flutter/flutter/issues/22998)
2. [Don't rely on global `pod` installation, use bundled version](https://github.com/flutter/flutter/issues/40135)

How to Test this PR?
-
1. Checkout the branch.
2. Create new project using `--use-bundler` via this command `flutter create --platform ios,android test_case --use-bundler`. It will make sure to add the proper `Gemfile` for you inside `ios` folder.
3.  Add new dependency to the `pubspec.yaml` file that have a native counterpart. For example `webview_flutter:`
4. Build the iOS project via this command to build using `Bundler`, `flutter build ios --use-bundler`

## Pre-launch Checklist



[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes
